components/compositing/build.rs

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+extern crate toml;
+
+use std::env;
+use std::fs::File;
+use std::io::{Read, Write};
+use std::path::Path;
+
+fn main() {
+    let lockfile_path = Path::new(&env::var("CARGO_MANIFEST_DIR").unwrap()).join("..").join("..").join("Cargo.lock");
+    let revision_file_path = Path::new(&env::var_os("OUT_DIR").unwrap()).join("webrender_revision.rs");
+
+    let mut lockfile = String::new();
+    File::open(lockfile_path).expect("Cannot open lockfile")
+        .read_to_string(&mut lockfile)
+        .expect("Failed to read lockfile");
+
+    match toml::from_str::<toml::value::Table>(&lockfile) {
+        Ok(result) => {
+            let packages = result.get("package").expect("Cargo lockfile should contain package list");
+
+            match *packages {
+                toml::Value::Array(ref arr) => {
+                    let source = arr
+                        .iter()
+                        .find(|pkg| pkg.get("name").and_then(|name| name.as_str()).unwrap_or("") == "webrender")
+                        .and_then(|pkg| pkg.get("source").and_then(|source| source.as_str()))
+                        .expect("webrender package does not conatin source to read revision");
```

Comment:
Does the build still work if you create an [override](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#overriding-dependencies) for webrender in Cargo.toml?
```
[patch."https://github.com/servo/webrender"]
webrender = { path = "../webrender/webrender" }
```

Reply:
I just tried locally by specifying above section in `components/composition/cargo.toml`, seems build success without failure. Not sure if I tried correct way to validate this.

Reply:
It belongs in the Cargo.toml in the root of the repository instead.

Reply:
My bad, thanks for let me know. 

Just tried and build fails with overriding, as updated cargo lock omits `source` property to read revision information.

Reply:
I think in that case we should let the build proceed with a warning that the webrender revision could not be determined, and store a value like `<unknown>`.

Reply:
agreed, I'll try to update implementation to reflect those cases. Appreciate for let me know.